.github/workflows/lint.yaml

```diff
@@ -0,0 +1,86 @@
+name: lint
+on:
+  push:
+    tags:
+      - v*
+    branches:
+      - main
+      - release-*
+  pull_request:
+    branches:
+      - main
+      - release-*
+
+# cancel the in-progress workflow when PR is refreshed.
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event_name == 'pull_request' && github.head_ref || github.sha }}
+  cancel-in-progress: true
+
+permissions:
+  contents: read
+
+jobs:
+  codespell:
+    name: codespell
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+      - name: codespell
+        uses: codespell-project/actions-codespell@master
+        with:
+          skip: .git,*.sum
+          ignore_words_list:
+          check_filenames: true
+          check_hidden: true
+  misspell:
+    name: misspell
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+      - name: misspell
+        uses: reviewdog/action-misspell@v1
+
+  golangci:
+    name: golangci-lint
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+      - uses: actions/setup-go@v5
+        with:
+          go-version-file: go.mod
+      - name: golangci-lint
+        uses: golangci/golangci-lint-action@v6
+        with:
+          # Required: the version of golangci-lint is required and must be specified without patch version: we always use the latest patch version.
+          version: v1.55
```

Comment:
many times we observe below w/ golangci-lint:
1. versions in makefile & actions script quickly gets diverged (solvable)
2. we may use relatively latest linter on older codebase/release which might force us upgrading unrelated dependencies in respective branches (may need small discussion)

I know it's in draft and only leaving pointers atm, btw, description say `markdown lint` and content tries to lint more than that, working title may need to change by the end.

Reply:
we should have a new file like `build.env` where we will have the version mentioned and if its used in multiple place we just need to source and use that